unityProject/Assets/demo.cs

```diff
@@ -1,22 +1,74 @@
-﻿using System.Collections;
+﻿using System;
+using System.Collections;
 using System.Collections.Generic;
-using UnityEngine;
 using System.Runtime.InteropServices;
+using UnityEngine;
 
 public class demo : MonoBehaviour {
-		// Import DLL (visp-demo.dll)
+
+    // Import DLL (visp-demo.dll)
     [DllImport("visp-demo", CallingConvention = CallingConvention.Cdecl, EntryPoint = "dot_prod")]
-		//Imported function dot_prod()
+    //Imported function dot_prod()
     public static extern double dot_prod(uint[] vec);
 
-		//vectors:
-		//a = {a1, a2, a3};
-		//b = {b1, b2, b3};
+    // Import DLL (visp-demo.dll)
+    [DllImport("visp-demo", CallingConvention = CallingConvention.Cdecl, EntryPoint = "passFrame")]
+    //Imported function passFrame()
+    public static extern void passFrame(byte[] bitmap, int height, int width);
+
+
+
+    public WebCamTexture webcamTexture;
+    public Color32[] data;
+
+    //vectors:
+    //a = {a1, a2, a3};
+    //b = {b1, b2, b3};
+
+    private uint[] vec = {1,2,3,1,2,3};
+
+    void Start()
+    {
+        webcamTexture = new WebCamTexture();
+        Renderer renderer = GetComponent<Renderer>();
+        renderer.material.mainTexture = webcamTexture;
+        data = new Color32[webcamTexture.width * webcamTexture.height];
+        webcamTexture.Play();
+
+        uint[] vec = { 1, 2, 3, 1, 2, 3 };
 
-		private uint[] vec = {1,2,3,1,2,3};
+        Debug.Log("Dot Product of the vectors is:");
+        Debug.Log(dot_prod(vec));
+    }
 
-		void Start () {
-			Debug.Log("Dot Product of the vectors is:");
-			Debug.Log(dot_prod(vec));
-		}
+    void Update()
+    {
+
+        passFrame(Color32ArrayToByteArray(webcamTexture.GetPixels32()), webcamTexture.height, webcamTexture.width);
+
+    }
+
+    private static byte[] Color32ArrayToByteArray(Color32[] colors)
+    {
+        if (colors == null || colors.Length == 0)
+            return null;
+
+        int lengthOfColor32 = Marshal.SizeOf(typeof(Color32));
+        int length = lengthOfColor32 * colors.Length;
+        byte[] bytes = new byte[length];
```

Comment:
Since we're converting the RGB to grey in this function itself, length should be color.Length (only one channel). So you're correct. Let me change it.